Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SpawnerComponent.h"
+#include "ROS2/Frame/ROS2FrameComponent.h"
+#include "ROS2/ROS2Bus.h"
+#include "ROS2/ROS2GemUtilities.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Spawnable/Spawnable.h>
+
+#include <ROS2/Utilities/ROS2Conversions.h>
+
+namespace ROS2
+{
+    ROS2SpawnerComponent::ROS2SpawnerComponent()
+    {
+        // TODO - currently causes errors on close. It is here to enable URDF spawning in default point.
+        // SpawnerInterface::Register(this);
```

Comment:
A Github issue should be made around this problem with registering with the SpawnerInterface and the TODO replaced with a link to the issue.